be/src/common/cow.h

```diff
@@ -258,7 +279,37 @@ class Cow {
 
     // cast the data as mutable ptr if it's mutable no matter it's mutable or immutable.
     // NOTE:  ptr's use_count will be added by 1, and this is not safe because the data may be shared with others.
-    MutablePtr as_mutable_ptr() const { return const_cast<Cow*>(this)->get_ptr(); }
+    // DCHECK added to catch potential misuse in debug builds.
+    MutablePtr as_mutable_ptr() const {
+#ifndef NDEBUG
+        uint32_t ref_count = use_count();
+        if (ref_count > 2) {
```

Comment:
why ref_count == 2 is valid too?

Reply:
`ref_count == 2` is already invalid and unexpected.

But this is meaningless for the current stage and may be optimized later step by step since I found there are still many logs even though this is a large value.

I can optimize this in the later pr.